src/Illuminate/Cache/FileLock.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Support\Carbon;
+
+class FileLock extends Lock
+{
+    /**
+     * The parent array file store.
+     *
+     * @var \Illuminate\Cache\FileStore
+     */
+    protected $store;
+
+    /**
+     * The lock file path.
+     *
+     * @var string
+     */
+    protected $path;
+
+    /**
+     * The extension that will be used for lock files.
+     *
+     * @var string
+     */
+    protected $extension = '.lock';
+
+    /**
+     * Create a new lock instance.
+     *
+     * @param  \Illuminate\Cache\FileStore  $store
+     * @param  string  $name
+     * @param  int  $seconds
+     * @param  string|null  $owner
+     * @return void
+     */
+    public function __construct($store, $name, $seconds, $owner = null)
+    {
+        parent::__construct($name, $seconds, $owner);
+
+        $this->store = $store;
+        $this->path = $this->store->getDirectory().'/'.$this->name.$this->extension;
+    }
+
+    /**
+     * Get lock file content if exists.
+     *
+     * @return string|null
+     */
+    private function getLockFileContent()
+    {
+        if ($this->exists()) {
+            return json_decode($this->store->getFilesystem()->get($this->path));
+        }
+    }
+
+    /**
+     * Write the lock file.
+     *
+     * @param  \Illuminate\Support\Carbon $now
+     * @return bool
+     */
+    private function writeLockFile($now)
+    {
+        $this->store->getFilesystem()->put($this->path, json_encode([
+            'owner' => $this->owner,
+            'expiresAt' => $this->seconds === 0 ? null : $now->copy()->addSeconds($this->seconds),
+        ]), true);
+
+        return true;
+    }
+
+    /**
+     * Attempt to acquire the lock.
+     *
+     * @return bool
+     */
+    public function acquire()
+    {
+        $now = Carbon::now();
+        $content = $this->getLockFileContent();
+
+        if (is_null($content)) {
```

Comment:
In between these two lines there is a race condition where another process could acquire the lock and then this process would overwrite application's lock. Therefore, this entire implementation isn't safe.

Reply:
You are right. Can you mark it as a draft pull request? I will modify the file access to make sure that is locked during the whole process, and add tests that assure it.

Reply:
@rogervila feel free to send in a new pr

Reply:
Thank you @driesvints :)